### PR TITLE
[3.11] Docs: Fix Sphinx annotations in Doc/library/ctypes.rst (#107672)

### PR DESCRIPTION
**Pull request by erlend-aasland**
(cherry picked from commit 71a7c96ffeb0d7fef06be3e57468896e030967a5)

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

----
:books: Documentation preview :books:: https://cpython-previews--107686.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/ctypes.rst [hunk 1]
```diff
@@ -482,9 +483,11 @@ a string pointer and a char, and returns a pointer to a string::
...
-If you want to avoid the ``ord("x")`` calls above, you can set the
-:attr:`argtypes` attribute, and the second argument will be converted from a
-single character Python bytes object into a C char::
+If you want to avoid the :func:`ord("x") <ord>` calls above, you can set the
+:attr:`~_FuncPtr.argtypes` attribute, and the second argument will be converted from a
+single character Python bytes object into a C char:
+
+.. doctest::
 
    >>> strchr.restype = c_char_p
```
Comment (CharlieZhao95) [inline Doc/library/ctypes.rst:517]: It seems that the lack of `testsetup` caused the test to fail here. See **main branch:** https://github.com/python/cpython/blob/a6675b1a597c67be972598ac8562883fabe48099/Doc/library/ctypes.rst?plain=1#L465-L475

Comment (erlend-aasland) [reply Doc/library/ctypes.rst:517]: I'm not sure it is worth the hassle to fix this backport.

**Comment by erlend-aasland** [comment]
Sorry for the noise!

**Comment by erlend-aasland** [comment]
I have no idea why this happened; my local branch is at `upstream/3.11`. Oh well, this backport is not that important. Let's just close it.

**Pull request closed without merging**